.github/workflows/lint-renovate-pr.yml

```diff
@@ -0,0 +1,73 @@
+name: Lint and Update renovate PR
+
+# configure trigger by pull request
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs: 
+  prepare:
+    if: github.event.pull_request.user.login == 'renovate[bot]'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Get build image from Makefile
+        id: build_image_step
+        run: echo "build_image=$(make print-build-image)" >> "$GITHUB_OUTPUT"
+    outputs:
+      build_image: ${{ steps.build_image_step.outputs.build_image }}
+
+  lint_and_update:
+    runs-on: ubuntu-latest
+    needs:
+      - prepare
+    container:
+      image: ${{ needs.prepare.outputs.build_image }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GH_BOT_ACCESS_TOKEN }}
+
+      - name: Run Git Config
+        run: |
+          git config --global --add safe.directory '*'
+          git config --global user.email "${{ github.event.pull_request.user.login }}@users.noreply.github.com"
+          git config --global user.name "${{ github.event.pull_request.user.login }}"
+  
+      - name: Install the gh cli
+        uses: ksivamuthu/actions-setup-gh-cli@v2
```

Comment:
Why don't we include `gh` into build image?

Reply:
we can, how about do that in a separate PR, debugged 3 hours for this file because of the container, let's keep it smaller.

Reply:
ok, it is just one line change, let's update this and see how it goes.